providers/firestore.js

```diff
@@ -0,0 +1,182 @@
+const { Provider } = require('klasa');
+const firebase = require('firebase-admin');
+const { FieldValue } = firebase;
+
+module.exports = class extends Provider {
+
+	constructor(...args) {
+		super(...args);
+		this.db = null;
+	}
+	/**
+	 * Initiaizes the firestore
+	 * @private
+	 */
+	async init() {
+		await firebase.initializeApp({
+			credential: firebase.credential.cert(this.client.options.providers.firestore.credentials),
+			databaseURL: this.client.options.providers.firestore.databaseURL
+		});
+
+		this.db = firebase.firestore();
+	}
+	/**
+	 *Checks if the table/collection exists
```

Comment:
Spacing. Also, you could've probably re-used the current jsdocs from, say, rethinkdb provider and tweaked them accordingly.